src/pages/Search/SearchPage.tsx

```diff
@@ -816,243 +847,137 @@ function SearchPage({route}: SearchPageProps) {
-                    title={translate('common.downloadFailedTitle')}
-                    prompt={translate('common.downloadFailedDescription')}
-                    isSmallScreenWidth={isSmallScreenWidth}
-                    onSecondOptionSubmit={() => setIsDownloadErrorModalVisible(false)}
-                    secondOptionText={translate('common.buttonConfirm')}
-                    isVisible={isDownloadErrorModalVisible}
-                    onClose={() => setIsDownloadErrorModalVisible(false)}
-                />
-                <ConfirmModal
-                    title={translate('customApprovalWorkflow.title')}
-                    isVisible={isDEWModalVisible}
-                    onConfirm={() => {
-                        setIsDEWModalVisible(false);
-                        openOldDotLink(CONST.OLDDOT_URLS.INBOX);
-                    }}
-                    onCancel={() => setIsDEWModalVisible(false)}
-                    prompt={translate('customApprovalWorkflow.description')}
-                    confirmText={translate('customApprovalWorkflow.goToExpensifyClassic')}
-                    shouldShowCancelButton={false}
-                />
-            </FullPageNotFoundView>
-        </ScreenWrapper>
+                            </DragAndDropConsumer>
+                            {ErrorModal}
+                        </DragAndDropProvider>
+                    ) : (
+                        <SearchPageWide
+                            queryJSON={queryJSON}
+                            searchResults={searchResults}
+                            searchRequestResponseStatusCode={searchRequestResponseStatusCode}
+                            isMobileSelectionModeEnabled={isMobileSelectionModeEnabled}
+                            headerButtonsOptions={headerButtonsOptions}
+                            footerData={footerData}
+                            selectedPolicyIDs={selectedPolicyIDs}
+                            selectedTransactionReportIDs={selectedTransactionReportIDs}
+                            selectedReportIDs={selectedReportIDs}
+                            latestBankItems={latestBankItems}
+                            onBulkPaySelected={onBulkPaySelected}
+                            handleSearchAction={handleSearchAction}
+                            onSortPressedCallback={onSortPressedCallback}
+                            scrollHandler={scrollHandler}
+                            initScanRequest={initScanRequest}
+                            PDFValidationComponent={PDFValidationComponent}
+                            ErrorModal={ErrorModal}
+                            shouldShowOfflineIndicator={shouldShowOfflineIndicator}
+                            offlineIndicatorStyle={offlineIndicatorStyle}
+                            shouldShowFooter={shouldShowFooter}
+                        />
+                    )}
+                    {(!shouldUseNarrowLayout || isMobileSelectionModeEnabled) && (
```

Comment:
This references my previous comment. I see a lot of the changes in the PR could be decoupled from the preloading itself - it looks like much of the performance gain might be coming from disabling a part of the React tree for the initial load. Maybe it can be integrated separately?

Reply:
@adhorodyski I’ll double-check with more tests, but it seems that decomposition brings most of the benefit, while preload adds less and might not be worth including. Removing preload bugs solutions also improves results, so decomposition alone already looks good. I’ll share detailed results tomorrow.